### ISSUE TITLE
No connection established

### ISSUE THREAD

**Issue by ddaanniiieeelll** [simoncozens/CritParty#7]
We tried CritParty today with two people and couldn't establish a connection.
After an initial error message "A tab needs to be open" nothing seemed to work.
We tried both ways, trying to host and trying to join a session but nothing happened.

There were no further pop up error messages, nor messages in the CritParty Window or the output window.
We both use Glyphs 2.6.4 (1286) which according to Glyphs is up to date.

**Comment by schriftgestalt** [comment]
Can you update to the latest cutting version (activate it in Preferences > Updates).

**Comment by ddaanniiieeelll** [comment]
We both did update to Version 2.6.5 (1339) but somehow there is still nothing happening.
Also no error messages I could share.

**Comment by simoncozens** [comment]
Daft question but... did you open a tab with some glyphs in it before trying again?

**Comment by simoncozens** [comment]
Also please try uninstalling, downloading the debug version from http://www.corvelsoftware.co.uk/software/CritParty.glyphsPlugin-debug.zip and trying again; then attach the log files from `Documents/webrtc_logs/` on each computer.

**Comment by ddaanniiieeelll** [comment]
> Daft question but... did you open a tab with some glyphs in it before trying again?

Yes I did. At first I was just in the overview but then I got a pop up error message and then of course I tried it with an open tab with a glyph in it, but that didn’t help.

> Also please try uninstalling, downloading the debug version from http://www.corvelsoftware.co.uk/software/CritParty.glyphsPlugin-debug.zip and trying again; then attach the log files from Documents/webrtc_logs/ on each computer.

We will as soon as we find the time again, thanks for the debug version :)

**Comment by gorjious** [comment]
A friend and I were unable to establish a connection. 
At first, we were on the different version, 2.6.6 (1343) and 2.6.5 (1342), then we tried on the same 2.6.5 (1342) version. In each case, it wasn't working. 

One thing that was strange was that whenever my friend would share a session ID, I would click connect and nothing would happen and no popup error messages. Whenever I would create a session ID and my friend tried to join, Glyphs would immediately crash and they would get the message "Host Closed Connection Party's over, folks". 

We were also unsure what should be the setup on each of our computers. Should only one person have a glyphs file open the other only glyphs open with no file loaded? Will a new window open for the other person? Should the other person have a blank file open? Should both had the same file? 

We tried multiple scenarios but no success. 

Any suggestions on what we can try?
If it's a firewall issue, how can we check that?

**Comment by simoncozens** [comment]
That's very weird. Could you try the debug version linked above? The host user needs to have a glyphs file open (the plugin will shout at them if they don't) and it doesn't matter about the guest user.

**Comment by gorjious** [comment]
We tried again with a new install of the debug version on Glyphs Version 2.6.5 (1342). 
First time I tried as Host with one tab open. Friend had nothing open. Result was that Glyphs crashed.
Second time my friend tried as Host and I had nothing open. Result was that nothing happened when I click join.

I've emailed you both of webrtc_logs and the glyphs error I received.

**Comment by simoncozens** [comment]
Brilliant. Thanks. I shall take a look.

**Comment by simoncozens** [comment]
Aargh, it's this problem: https://groups.google.com/forum/#!msg/discuss-webrtc/P4vVIvPL7Tg/nDxXaxjbCQAJ
Strings are getting lost within the WebRTC library, somewhere between Objective C and C++. And I have no idea how to solve it. :-(

**Comment by gorjious** [comment]
Arg! Hopefully, there is some feedback on the forum on how to solve it. 
If there is an update, feel free to contact me and to help test it.

**Comment by schriftgestalt** [comment]
The c++ <> objectiveC Problem sounds like a memory issue. How do you convert the string. Maybe the source string is deallocated before the target gets an hold of it?

**Comment by simoncozens** [comment]
The strings are being created here: https://github.com/simoncozens/CritParty/blob/d8dcd54549acceab4d680493e121bf86c4d60b04/CritParty/CritParty/CritParty.m#L433-L438

and then gets to C++ through this code here: https://chromium.googlesource.com/external/webrtc/+/branch-heads/62/webrtc/sdk/objc/Framework/Classes/PeerConnection/RTCPeerConnection.mm#216

**Comment by schriftgestalt** [comment]
The strings are all in the source, so they shouldn’t be deallocated. I don’t understand the code, yet. Can you run it in Xcode with enabled memory protection and zombie settings?

**Comment by simoncozens** [comment]
How do I debug a plug-in in Xcode?

**Comment by gorjious** [comment]
Has CritParty been updated for Glyphs 3? 

Tried using the plugin again with a friend but we were still running into connection issues.

As soon as either one of us tried to Join by entering the Session ID and Password, it immediately disconnects and says something along the lines of "the party is over".

We were both on Glyphs 3 Cutting Edge 3.0.5 (3131) with installing the latest version of the plugin
